src/logic/currentSession/hooks/useCurrentSafeAddressSync.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+import { extractPrefixedSafeAddress, history } from 'src/routes/routes'
+import addCurrentSafeAddress from 'src/logic/currentSession/store/actions/addCurrentSafeAddress'
+import { useDispatch } from 'react-redux'
+import { Dispatch } from 'redux'
+
+export const useCurrentSafeAddressSync = (): void => {
+  const dispatch = useDispatch<Dispatch>()
+
+  useEffect(() => {
+    dispatch(addCurrentSafeAddress(extractPrefixedSafeAddress(history.location.pathname).safeAddress))
+
+    const unsubscribe = history.listen((location) => {
```

Comment:
Can't we just use the [`useLocation`](https://v5.reactrouter.com/web/api/Hooks/uselocation) hook?
Docs are v5 but AFAIK it exists in v4 as well.

Reply:
True, we can simplify this hook a lot by using `useLocation`. Thanks!